src/__tests__/global-object/global-object.test.tsx

```diff
@@ -28,6 +28,42 @@ const expectedGlobals_GIPClientSide = {
 
 describe('Global object', () => {
   describe('getServerSideProps', () => {
+    describe('useDocument=true', () => {
+      describe.each(['initial', 'client'])('%s render', (renderType) => {
+        it("executes page's exports with expected env globals", async () => {
+          const initialRoute = renderType === 'client' ? '/' : '/ssr';
+          const { page } = await getPage({
+            nextRoot: path.join(__dirname, '__fixtures__'),
+            route: initialRoute,
+            useDocument: true,
+          });
+          const { container } = render(page);
+
+          // Client side navigation to SSR page
+          if (renderType === 'client') {
+            userEvent.click(screen.getByText('Go to SSR'));
+            await screen.findByText('Page');
+          }
+
+          const head = container.querySelector('head') as HTMLHeadElement;
+          expect(
+            JSON.parse(
+              (head.querySelector(
+                'meta[name="global-object"]'
+              ) as HTMLMetaElement).content
+            )
+          ).toEqual({
+            importTime_document: false,
+            importTime_window: false,
+            gip_runTime_window: false,
+            gip_runTime_document: false,
+            component_runTime_window: true, // TODO: should be false
+            component_runTime_document: true, // TODO: should be false
```

Comment:
Not sure if possible because `render` is called after `getPage` by the `@testing-library/react`